### PR TITLE
Rename TraceConfiguration to TraceOptions to match Options in the other libs.

### PR DESCRIPTION
**Pull request by iantalarico**
FYI, The swapping of TraceConfiguration to TraceOptions didn't merge well.

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs [hunk 1]
```diff
@@ -289,11 +289,11 @@ public AbstractTraceTestApplication()
 
         public void ConfigureServices(IServiceCollection services)
         {
-            var traceConfig = TraceConfiguration.Create(GetSampleRate(), GetBufferOptions());
+            var traceConfig = Common.TraceOptions.Create(GetSampleRate(), GetBufferOptions());
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs]: Nit: `traceOptions`?

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/InternalTraceOptions.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/InternalTraceOptions.cs]: 2017

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceOptions.cs [hunk 3]
```diff
@@ -12,25 +12,42 @@
...
 
         /// <summary>
         /// Creates a <see cref="TraceOptions"/>.
         /// </summary>
-        /// <param name="shouldTrace">True if the tracing should occur.</param>
-        public static TraceOptions Create(bool shouldTrace) => new TraceOptions(shouldTrace);
+        /// <param name="qpsSampleRate">Optional, the qps sample rate.  The sample rate determines
+        ///     how often requests are automatically traced. Defaults to <see cref="DefaultQpsSampleRate"/>
+        /// </param>
+        /// <param name="bufferOptions">Optional, the buffer options.  Defaults to a <see cref="BufferType.Timed"/></param>
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceOptions.cs]: Nit: end these two descriptions with '.'

**Comment by iantalarico** [comment]
Fixed, thanks Mike!

**Pull request merged**